src/AWS/Orleans.Streaming.SQS/Hosting/ClientBuilderExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Orleans.Configuration;
+using Orleans.Hosting;
+using OrleansAWSUtils.Streams;
+
+namespace Orleans.Hosting
+{
+    public static class ClientBuilderExtensions
+    {
+        /// <summary>
+        /// Configure cluster client to use SQS persistent streams.
+        /// </summary>
+        public static IClientBuilder AddSqsStreams(this IClientBuilder builder, string name, Action<SqsStreamOptions> configureOptions)
+        {
+            return builder.ConfigureServices(services => services.AddClusterClientSqsStreams(name, configureOptions));
+        }
+
+        /// <summary>
+        /// Configure cluster client to use SQS persistent streams.
+        /// </summary>
+        public static IClientBuilder AddSqsStreams(this IClientBuilder builder, string name, Action<OptionsBuilder<SqsStreamOptions>> configureOptions = null)
+        {
+            return builder.ConfigureServices(services => services.AddClusterClientSqsStreams(name, configureOptions));
+        }
+
+        /// <summary>
+        /// Configure cluster client to use SQS persistent streams.
+        /// </summary>
+        public static IServiceCollection AddClusterClientSqsStreams(this IServiceCollection services, string name, Action<SqsStreamOptions> configureOptions)
+        {
+            return services.AddClusterClientSqsStreams(name, ob => ob.Configure(configureOptions));
+        }
+
+        /// <summary>
+        /// Configure cluster client to use SQS persistent streams.
+        /// </summary>
+        public static IServiceCollection AddClusterClientSqsStreams(this IServiceCollection services, string name,
```

Comment:
I think we should *not* add any extension methods which hang from `IServiceCollection` and we should endeavor to remove the existing ones.

Reply:
Because otherwise we will have methods which are ambiguous between client & silo & SignalR & ASP.NET. I'm thinking about the future, but the issue is present here today: we need to name the methods differently for client and silo instead of having a nice simple "AddSqsStreams" method.

Reply:
I see... Well, other framework (including asp.net) does that IIRC. However, I think we should have diff assemblies for client and silo for the providers. For example, in Membership, makes no sense to bring the Membership table to the client, but instead it should be only have the gateway. So `Orleans.Clustering.XXXX` and `Orleans.Clustering.XXXX.Client`(or `Gateway`). I know, extra nuget package to maintain, but the benefits are:

1. Avoid this confusion with extensions
2. Avoid having 2 methods for client and silo
3. In the case of clustering, remove the dependency on the runtime/silo assemblies on teh client.

Reply:
Will discuss then address in separate PR based on consensus.

Reply:
Maybe the extension methods should at least live in their own namespace? One for the clientbuilder and one for the silo? Better for discoverability.